### DISCUSSION TITLE
getting import error

### DISCUSSION THREAD

**Discussion by Lucian-Ryu** [Cyfrin/foundry-full-course-cu#357] [Q&A]
hello, i just started the second video of the course, and when i created my new project folder ```foundry-fund-me-f23``` first it doesnt let me use ``` forge init``` and i need to use ``` forge init --force``` to initialize the project. 
And when i open it  in a new VS code i cant

 import  ``` "forge-std/Test.sol"``` neither ```"forge-std/Script.sol";``` from the files that were automatically created.

i get this error message 
```
Source "forge-std/Script.sol" not found: File not found. Searched the following locations: "".solidity(6275)
```

any idea whats happening?

thank you 

![Screenshot 1](https://github.com/Cyfrin/foundry-full-course-f23/assets/114004261/6b08acd0-57b2-4472-950b-3390cda233c5)
![Screenshot 2](https://github.com/Cyfrin/foundry-full-course-f23/assets/114004261/82691215-0203-431e-9e1a-5f785a8c5744)
![Screenshot 3](https://github.com/Cyfrin/foundry-full-course-f23/assets/114004261/59953976-da7a-43bf-b1fa-193f50edf269)

**Comment by shikhar229169** [comment]
Can you restart the vs code and try again.
But if that doesn't work, try it in a new folder.

**Reply by Lucian-Ryu** [reply]
This didn’t solve the issue unfortunately.

**Reply by Lucian-Ryu** [reply]
I could get rid of the error by passing the command —no-commit. 
Apparently when you forge init it also creates a repository in git. And somehow I already had one. I don’t know why Patrick didn’t get that error. 
I’ll  continue studying today. Thanks for your help.

**Reply by thurendous** [reply]
Yes, you are right. Because I think there are several folders which are under git management. And that is the reason why you need to add `--no-commit` to ignore the current unstaged change.

**Comment by ghost** [comment]
Hello, @Lucian-Ryu, I also encountered a similar issue recently.

When I executed the `forge init` command, it generated the `lib` folder, but it turned out to be empty. This empty `lib` folder caused the errors I was facing.

To resolve this, I ran the `forge init --force` command. This command forcefully creates the necessary folders within the `lib` folder. After running it, I restarted Visual Studio Code, and everything started working fine again.

**Comment by Lucian-Ryu** [comment]
Thanks for the message, I also run forge unit —force

But I was still getting error. Someone told me that when you run forge init it automatically creates a git repo, but that somehow we already had one.

The way I could manage to not show me any error was to run —no-commit command.

That’s as far as I could go, because of work I will not be able to continue the course till tomorrow. Let’s see what happens.

Cheers

**Reply by ghost** [reply]
Try to create a new demo foundry project and check if you get the same error.

**Reply by ghost** [reply]
Go through the issue #201 you might find the solution there :)

**Comment by Ilya88Weber** [comment]
I had the same exact issue and my solution is: 
'''import "../lib/forge-std/src/Script.sol";
import {SimpleStorage} from "../src/SimpleStorage.sol; '''

**Comment by theirrationalone** [comment]
Try this
```terminal
forge init --no-commit
```

**Reply by theirrationalone** [reply]
Actually solution was already in your error logs... 


![Screenshot_2023-08-05-23-41-21-225_com.android.chrome.jpg](https://github.com/Cyfrin/foundry-full-course-f23/assets/96755510/e26d32fd-9751-4b49-9872-b2cd8d99c0fb)

**Comment by autherix** [comment]
Hi there,
I've experienced both these issues, and I want to explain it once for new commers like me 

## The first error:
```
The target directory is a part of or on its own an already initialized git repository,
and it requires clean working and staging areas, including no untracked files.

Check the current git repository's status with `git status`.
Then, you can track files with `git add ...` and then commit them with `git commit`,
ignore them in the `.gitignore` file, or run this command again with the `--no-commit` flag.

If none of the previous steps worked, please open an issue at:
https://github.com/foundry-rs/foundry/issues/new/choose
```
### Root Cause
Basically when you run `forge init` inside a directory, it initializes a git repository in that directory (if it's not already in a git repo), then creates a new commit to that repo containing all the files for the initialized foundry project
#### So If you run this command in a folder which is inside a git repository
  - If current repository's staging area is not clean
  - Or there is any untracked files  

### It gives us this error, which could be fixed by:
- Cleaning the staging area & Removing untracked files from scope, which is possible by:
  - moving them to the .gitignore zone
  - or deleting those changes
- Or using the switch `--no-commit`, so we tell <b>forge</b> not to try to create a new commit (it still initializes a git repo if not already created), <i>In this case, you should consider to add changes & commit by yourself</i>

## Second Error:
Starting from Lesson 6, we need to use this code
```solidity
import "forge-std/Script.sol";
```
This is obviously needed for us to make our contract inherit from the Script library from the forge standard libraries...
But in vscode when using the solidity extension (hardhat), there will be an error saying:
```
Source "forge-std/Script.sol" not found: File not found. Searched the following locations: "".solidity(6275)
```
But I suppose this is not an issue with the code, or library or terminal or foundry, but it's because of the extension itself
I've run the command `forge script script/DeploySimpleStorage.s.sol` and it worked correctly, so the error just remains to mess with us !

One way to solve it is to use relational addressing like when importing local modules:
```solidity
import {Script} from "../lib/forge-std/src/Script.sol";
```
But I didn't like that, so going on with a red line on editor !!

**Reply by theirrationalone** [reply]
relational pathing could be the solution.... 

But the problem is vscode itself.... Actually vscode's intelli-sense couldn't be able to sense `paths` on that slot of time.... Two reasons i can guess.... 


1. Vscode is outdated Or user messed something `UP` with new update auto installation orrrr there's a bug in new update. Due to which `service workers` unable to work. 

2. User's machine isn't capable of doing so much.... That's vscode is taking time to index all new entries into that project. Due to which `service workers` pushing entries into a `thread` and `thread` is puting emtries into a `queue`. And by doing `relational pathing` you actually cutting down AI's work. 

So now everybody should got to know what mitigations could be possible.
